src/modules/fancyzones/FancyZonesLib/WindowDrag.cpp

```diff
@@ -151,23 +151,18 @@ void WindowDrag::SwitchSnappingMode(bool isSnapping)
 {
     if (!m_snappingMode && isSnapping) // turn on
     {
-        m_highlightedZones.Reset();
         SetWindowTransparency();
 
         if (FancyZonesSettings::settings().showZonesOnAllMonitors)
         {
             for (const auto& [_, workArea] : m_activeWorkAreas)
             {
-                if (workArea)
+                if (workArea && workArea.get() != m_currentWorkArea)
```

Comment:
everything works now, but I don't get this change. Where is m_currentWorkArea being shown now?

Reply:
It's for showing non-activated layouts on all other monitors when the `Show zones on all monitors while dragging` option is activated. 
Zones on the active one are updated in the `MoveSizeUpdate`.